components/odh-notebook-controller/controllers/notebook_webhook.go

```diff
@@ -253,8 +265,103 @@ func (w *NotebookWebhook) Handle(ctx context.Context, req admission.Request) adm
 	return admission.PatchResponseFromRaw(req.Object.Raw, marshaledNotebook)
 }
 
+func (w *NotebookWebhook) ClusterWideProxyIsEnabled() bool {
+	proxyResourceList := &configv1.ProxyList{}
+	err := w.Client.List(context.TODO(), proxyResourceList)
+	if err != nil {
+		return false
+	}
+
+	for _, proxy := range proxyResourceList.Items {
+		if proxy.Name == "cluster" {
+			if proxy.Spec.HTTPProxy != "" && proxy.Spec.HTTPSProxy != "" &&
+				proxy.Spec.NoProxy != "" && proxy.Spec.TrustedCA.Name != "" {
+				return true
+			}
+		}
+	}
+	return false
+
+}
+
 // InjectDecoder injects the decoder.
 func (w *NotebookWebhook) InjectDecoder(d *admission.Decoder) error {
 	w.decoder = d
 	return nil
 }
+
+func InjectProxyConfig(notebook *nbv1.Notebook) error {
+	notebookContainers := &notebook.Spec.Template.Spec.Containers
+	var imgContainer corev1.Container
+
+	for _, container := range *notebookContainers {
+		// Update notebook image container
+		if container.Name == notebook.Name {
+			var newVars []corev1.EnvVar
+			imgContainer = container
+
+			for key, val := range proxyEnvVars {
+				keyExists := false
+				for _, env := range imgContainer.Env {
+					if key == env.Name {
+						keyExists = true
+					}
+				}
+				if !keyExists {
+					newVars = append(newVars, corev1.EnvVar{Name: key, Value: val})
+				}
+			}
+
```

Comment:
Does the value come from Proxy cluster? it does set ""
~~~
        - name: HTTPS_PROXY
        - name: HTTP_PROXY
        - name: NO_PROXY
~~~